docs/4.0. Behaviour - Authorization Server.md

```diff
@@ -6,66 +6,84 @@ _(c) AMWA 2019, CC Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)_
-in [RFC 4716][RFC-4716].
+The format of the keys SHOULD be the JSON Web Key (JWK) format described in Section 4 of [RFC 8414][RFC-8414] and
+SHOULD form a JSON Web Key Set (JWKS) described in Section 5 of [RFC 8414][RFC-8414]. The Authorization Server MAY
+present more than one key within a JSON Web Key Set, with each key being an entry in the "keys" array.
 
-Any public keys used to sign tokens for NMOS APIs MUST use a `kid` (Key ID) matching the following regular expression.
-The trailing integer in this expression is intended to be an integer timestamp matching the issue time of the key.
+Any JSON web Keys used to sign tokens for NMOS APIs MUST use a `kid` (Key ID) matching the following regular
+expression. The trailing integer in this expression is intended to be an integer timestamp matching the issue time of
+the key to facilitate key rotation.
 ```
 ^x-nmos-\d+$
 ```
 
 ### Changing Keys
 
 When transitioning to a new public/private key pair used for signing tokens the Authorization Server SHOULD provide
-both the old and new public key at the `/jwks` endpoint until all tokens that may be verified by the old public key
+both the old and new public key at the public key endpoint until all tokens that may be verified by the old public key
 would have expired. However, if a private key is known to be compromised, the Authorization Server MUST remove it
-from the `/jwks` endpoint immediately.
+from the public key endpoint immediately.
 
-Authorization Servers SHOULD provide new public keys on the `/jwks` endpoint for at least 2 hours before issuing tokens
-signed by the corresponding private key to allow time for clients to cache the new public key.
+Authorization Servers SHOULD provide new public keys at the public key endpoint endpoint for at least 2 hours before
+issuing tokens signed by the corresponding private key to allow time for clients to cache the new public key.
 
 ## Client Registration
 
-The Authorization Server MUST NOT accept unregistered clients.
+The Authorization Server MUST NOT grant tokens to unregistered clients.
 
 Authorization Servers SHOULD support a manual mechanism for registering clients (e.g an HTML web form) allowing the
 client type and redirect URIs to be provided.
 
+Clients and Authorization Servers MUST provide for the OAuth 2.0 Dynamic Client Registration Protocol
+[RFC 7591][RFC-7591] to allow dynamic registration of clients to the Authorization Server to allow for
+zero-configuration setup. This endpoint MUST be protected by a transport-layer security mechanism.
+
+Whilst it is RECOMMENDED that clients provide a means of authorization when registering dynamically (which is beyond
+the scope of this specification), Section 3 of [RFC 7591][RFC-7591] permits that clients may register with no
+authorization. The client registration endpoint SHOULD be rate-limited to prevent Denial-of-Service attacks.
```

Comment:
Is this DoS caveat mentioned in the dynamic registration spec? It seems unusual to add it just to this resource when the whole API could be targeted.

Reply:
Yes its listed in RFC 7591